jetty-maven/jetty-maven-core/src/main/java/org/eclipse/jetty/maven/AbstractWebAppMojo.java

```diff
@@ -0,0 +1,461 @@
+    protected PluginDescriptor plugin;
+    
+    /**
+     * The project's remote repositories to use for the resolution.
+     */
+    @Parameter (defaultValue = "${project.remoteArtifactRepositories}", readonly = true, required = true)
+    private List<ArtifactRepository> remoteRepositories;
+
+    /**
+     * 
+     */
+    @Component
+    private RepositorySystem repositorySystem;
+    
+    /**
+     * The current maven session
+     */
+    @Parameter (defaultValue = "${session}", required = true, readonly = true)
+    private MavenSession session;
+    
+    /**
+     * Default supported project type is <code>war</code> packaging.
+     */
+    @Parameter
+    protected List<String> supportedPackagings = Collections.singletonList("war");
+    
+    /**
+     * List of deps that are wars
+     */
+    protected List<Artifact> warArtifacts;
+    
+    /**
+     * Webapp base before applying overlays etc
+     */
+    protected Resource originalBaseResource;
+    
+    /**
+     * List of jars with scope=provided
+     */
+    protected List<File> providedJars;
+
+    /**
+     * System properties from both systemPropertyFile and systemProperties.
+     */
+    protected Map<String, String> mergedSystemProperties;
+
+    @Parameter(defaultValue = "${mojoExecution}", readonly = true)
+    protected MojoExecution mojo;
+
+    @Inject
+    private MavenPluginManager mavenPluginManager;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException
+    {
+        ClassRealm original = (ClassRealm)Thread.currentThread().getContextClassLoader();
+
+        Plugin targetPlugin = MojoExecutor.plugin("org.eclipse.jetty." + getEnvironment(),
+                "jetty-" + getEnvironment() + "-maven-plugin",
+                plugin.getVersion());
+
+        targetPlugin.setDependencies(plugin.getPlugin().getDependencies());
+        RepositorySystemSession repositorySystemSession = session.getRepositorySession();
+        List<RemoteRepository> remoteRepositories = session.getCurrentProject().getRemotePluginRepositories();
+
+        try
+        {
+            PluginDescriptor pluginDescriptor =
+                    mavenPluginManager.getPluginDescriptor(targetPlugin, remoteRepositories, repositorySystemSession);
+            ClassRealm classRealm = buildPluginManager.getPluginRealm(session, pluginDescriptor);
+
+            Xpp3Dom originalConfiguration = mojo.getConfiguration();
+
+            Xpp3Dom newConfiguration = new Xpp3Dom(originalConfiguration.getName());
+            for (Xpp3Dom child : originalConfiguration.getChildren())
+            {
+                //remove environment mojo
+                if (!"mojo".equals(child.getName()) && !"environment".equals(child.getName()))
+                {
+                    newConfiguration.addChild(child);
+                }
+            }
+
+            Thread.currentThread().setContextClassLoader(classRealm);
+            MojoExecutor.executeMojo(
```

Comment:
IIUC what's happening here, we're copying the configuration from the core UnassembledWebAppMojo and AbstractWebAppMojo onto the UnassembledWebAppMojo and AbstractWebAppMojo that are in the specific eeX plugin version?

If so, I don't like having those core classes duplicated in every eeX module - they must be exactly identical in terms of declared parameters otherwise configuration info is missed and can cause jetty to fail in unexpected and hard to detect kind of ways.

Reply:
please read my comment here https://github.com/eclipse/jetty.project/pull/9216#discussion_r1091387036 about `LoginService` without this field which is already sort of deprecated in jetty-10.0.x (https://github.com/eclipse/jetty.project/blob/ca19b143a9de1c222b517028afc118482f7c093b/jetty-maven-plugin/src/main/java/org/eclipse/jetty/maven/plugin/AbstractWebAppMojo.java#L242) we could get rid of a lot duplicated classes.

Reply:
And BTW I explained in my comment why we can still have duplicated classes (frankly there are already so many ;) ) https://github.com/eclipse/jetty.project/pull/9216#issuecomment-1414563027
BUT the most important is to have this extra layer on the top and this unique groupId/artifactId which is backward compatible.

Reply:
and this PR is already removing some duplication :)
<img width="1760" alt="Screenshot 2023-02-03 at 11 32 59 am" src="https://user-images.githubusercontent.com/19728/216489972-5430a71c-539b-4fdb-bc8a-1bd99aad4738.png">